tests/entrypoints/test_openai_server.py

```diff
@@ -250,5 +310,243 @@ async def test_batch_completions(server, client: openai.AsyncOpenAI):
+    }]
+    chat_completion = await client.chat.completions.create(
+        model=MODEL_NAME,
+        messages=messages,
+        max_tokens=10,
+        extra_body=dict(
+            guided_choice=TEST_CHOICE
+        )
+    )
+    choice1 = chat_completion.choices[0].message.content
+    assert choice1 in TEST_CHOICE
+
+    messages.append({"role": "assistant", "content": choice1})
+    messages.append({
+        "role": "user",
+        "content": "I disagree, pick another one"
+    })
+    chat_completion = await client.chat.completions.create(
+        model=MODEL_NAME,
+        messages=messages,
+        max_tokens=10,
+        extra_body=dict(
+            guided_choice=TEST_CHOICE
+        )
+    )
+    choice2 = chat_completion.choices[0].message.content
+    assert choice2 in TEST_CHOICE
+    assert choice1 != choice2
+
+
+async def test_guided_decoding_type_error(server, client: openai.AsyncOpenAI):
+    with pytest.raises(openai.BadRequestError):
+        _ = await client.completions.create(
+            model=MODEL_NAME,
+            prompt="Give an example JSON that fits this schema: 42",
+            extra_body=dict(
+                guided_json=42
+            )
+        )
+
+    messages = [{
+        "role": "system",
+        "content": "you are a helpful assistant"
+    }, {
+        "role": "user",
+        "content": "The best language for type-safe systems programming is "
+    }]
+    with pytest.raises(openai.BadRequestError):
+        _ = await client.chat.completions.create(
+            model=MODEL_NAME,
+            messages=messages,
+            extra_body=dict(
+                guided_regex={1: "Python", 2: "C++"}
+            )
+        )
+
+    with pytest.raises(openai.BadRequestError):
+        _ = await client.completions.create(
+            model=MODEL_NAME,
+            prompt="Give an example string that fits this regex",
+            extra_body=dict(
+                guided_regex=TEST_REGEX,
+                guided_json=TEST_SCHEMA
+            )
+        )
+
+
+def test_guided_logits_processors():
+    """Basic unit test for RegexLogitsProcessor and JSONLogitsProcessor."""
+    from vllm.model_executor.guided_logits_processors import (
+        RegexLogitsProcessor,
+        JSONLogitsProcessor
+    )
+    from transformers import AutoTokenizer
+    from torch import rand
+    
+    tokenizer = AutoTokenizer.from_pretrained('HuggingFaceH4/zephyr-7b-beta')
+    regexLP = RegexLogitsProcessor(TEST_REGEX, tokenizer)
+    jsonLP = JSONLogitsProcessor(TEST_SCHEMA, tokenizer)
+    tensor = rand(32000)
+
+    regexLP.init_state()
+    token_ids = tokenizer.encode(
+        f"Give an example IPv4 address with this regex: {TEST_REGEX}")
+    assert regexLP(token_ids, tensor).shape == tensor.shape
+    
+    jsonLP.init_state()
+    token_ids = tokenizer.encode(
+        f"Give an employee profile that fits this schema: {TEST_SCHEMA}")
+    assert jsonLP(token_ids, tensor).shape == tensor.shape
```

Comment:
currently this tests the output tensor has the same shape as input tensor
we should also test that output tensor does not equal input tensor. `assert sonLP(token_ids, tensor) == tensor` this will always be true because `tensor` is modified in place. so you do need to make a copy first, also need to use torch.allclose or something to verify `equals`.